### ISSUE TITLE
Can't create new documents due to a `/` at start of filename

### ISSUE THREAD

**Issue by jonathanyeong** [tinacms/tinacms#4543]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I'm unable to create new documents in Tina Cloud. I see the following error:

<img width="511" alt="Screenshot 2024-05-23 at 9 30 59 AM" src="https://github.com/tinacms/tinacms/assets/3861088/31c2d10e-b5eb-417a-8e46-5d6fbb7ad377">

Here's the mutation:
<img width="1192" alt="Screenshot 2024-05-23 at 9 33 59 AM" src="https://github.com/tinacms/tinacms/assets/3861088/d3e79162-48fa-4b75-8b33-79448ef3f1fd">

After debugging with @kldavis4 in Discord, they sent an error on the API
```
ERROR    HttpError: path cannot start with a slash: {"resource":"Commit","field":"path","code":"invalid"}
```

My content is hosted in a [separate repo](https://github.com/jonathanyeong/personal-website-content). Tina path is `""` since I don't want to nest my content into another folder.

I'm able to save documents locally. And I'm able to edit documents on Tina Cloud.

### Reproduction

https://github.com/jonathanyeong/personal-website

### Steps to reproduce

1. Go to Tina Cloud
2. Create a new document
3. Fill in the required fields
4. Click Save

### System Info

```shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 1.39 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.4.0 - ~/.asdf/installs/nodejs/21.4.0/bin/node
    Yarn: 1.22.22 - ~/Code/personal-website/node_modules/.bin/yarn
    npm: 10.8.0 - ~/.asdf/plugins/nodejs/shims/npm
    pnpm: 8.12.1 - ~/.asdf/installs/nodejs/21.4.0/bin/pnpm
  Browsers:
    Chrome: 124.0.6367.210
    Safari: 17.5
  npmPackages:
    @tinacms/cli: ^1.5.45 => 1.5.45
    tinacms: ^1.6.3 => 1.6.3
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by kldavis4** [comment]
Move your content to `content/posts` - there is an issue with setting path on the collection to an empty string.

Also, unrelated but delete `tina/_generated_` from the content repo (not needed) and copy the `tina/config.ts` from the site repo or it won't index

**Comment by kldavis4** [comment]
cc @ncn-ssw

**Comment by jonathanyeong** [comment]
Yup, moving the content to `/posts` or `/blog` folder works. Ideally I’d like to keep it flat since Astro has its own post schema that I'd like to keep in the main repo and out of the submodule.

I noticed in your [PR](https://github.com/jonathanyeong/personal-website-content/pull/3/files#diff-2a34f2130caf88af57a0a8ba6ce9aadaf7335a87e1280095cb5036850020b1d1) you included the `tina/config.ts` in the content repo. Couple of questions:
- In the [docs](https://tina.io/guides/tinacms/separate-content-repo/guide/), it mentions "With this approach you can continue to define your Tina config in your "website repo", but documents will be saved elsewhere.". That makes me think that I only need the `tina/config.js` file in the website repo and not the content repo. Is that false?
- If I'm duplicating the `tina/config.js` file in both the content and website repo, what should the path be? Should it be `src/content/posts` (assuming I move the content in the content repo into `/posts`). Or would I leave it empty like in your PR?

**Comment by jonathanyeong** [comment]
Looks like I can't use an `.` as the path either:

![CleanShot 2024-05-23 at 19 53 48@2x](https://github.com/tinacms/tinacms/assets/3861088/d5d8113f-dbdd-4583-a633-5dbc2129a8b3)

**Comment by jonathanyeong** [comment]
Sorry @kldavis4 I got super curious about what was happening 😅. I hope it's okay, I did a bit of an investigation! 

---

[This line](https://github.com/tinacms/tinacms/blob/3bf237a3eb53395651b1facb5d309350291d9d27/packages/tinacms/src/admin/pages/CollectionCreatePage.tsx#L46) in `CollectionCreatePage` adds the forward slash to the path if it's not a folder. The slash is then passed all the way down to the Github call which ends up failing because `gh: path cannot start with a slash (HTTP 422)`. Which makes sense since Octokit under the hood calls the [Create or Update file contents endpoint](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) which will result in a PUT path that looks something like `/repos/OWNER/REPO/contents//FILENAME.md` (note the extra slash)

If I modify the line in CollectionCreatePage to the one below, the mutation to create a document doesn't include the forward slash.

```
const appendFolder = folder && !filename.startsWith('/') ? `/${folder}/` : ''
```

I'm happy to put up a PR, but I wanted to double-check if my reasoning is valid. Is there some context I'm missing as to why the `/` was added in the `appendFolder` const?

If you disagree with the change, happy to close out this ticket! But it would be nice to support empty collection paths with Github.
